Sources/SwiftSyntaxBuilderGeneration/main.swift

```diff
@@ -0,0 +1,199 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import SwiftSyntax
+import SwiftSyntaxBuilder
+
+private let format = Format(indentWidth: 2)
+
+private let copyrightHeader = """
+    //// Automatically Generated
+    //// Do Not Edit Directly!
+    //===----------------------------------------------------------------------===//
+    //
+    // This source file is part of the Swift.org open source project
+    //
+    // Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+    // Licensed under Apache License v2.0 with Runtime Library Exception
+    //
+    // See https://swift.org/LICENSE.txt for license information
+    // See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+    //
+    //===----------------------------------------------------------------------===//
+
+
+    """
+
+let tokensFile = SourceFile {
+  ImportDecl(importTok: TokenSyntax.import.withLeadingTrivia(.docLineComment(copyrightHeader)), path: "SwiftSyntax")
+
+  ExtensionDecl(extendedType: "TokenSyntax", modifiersBuilder: { TokenSyntax.public.withLeadingTrivia(.newlines(1) + .docLineComment("/// Namespace for commonly used tokens with default trivia.") +
+    .newlines(1)) }, membersBuilder: {
+      for token in SYNTAX_TOKENS {
+        if token.isKeyword {
+          VariableDecl(letOrVarKeyword: .var,
+                       modifiersBuilder: {
+            if let text = token.text {
+              TokenSyntax.static.withLeadingTrivia(.newlines(1) + .docLineComment("/// The `\(text)` keyword") + .newlines(1))
+            } else {
+              TokenSyntax.static
+            }
+          },
+                       bindingsBuilder: {
+            // We need to use `CodeBlock` here to ensure there is braces around.
+
+            let accessor = CodeBlock {
+              FunctionCallExpr(calledExpression: MemberAccessExpr(base: "SyntaxFactory", name: "make\(token.name)Keyword"),
+                               leftParen: .leftParen,
+                               rightParen: .rightParen)
+
+              if token.requiresLeadingSpace {
+                FunctionCallExpr.leadingTriviaFunction
```

Comment:
I can’t really tell why but `FunctionCallExpr.leadingTriviaFunction` doesn’t sound like it generates a `withLeadingTrivia` call to me. I think it’s mostly because I wouldn't expect anything SwiftSyntaxBuilder code generation builder-specific to be declared on `FunctionCallExpr`. I think I’d prefer to have a free function like `createWithLeadingTriviaCall` to be declared at the top of the file.

Reply:
Moved all extensions to the top as a free function